### DISCUSSION TITLE
Immediate GET after a PUT with Minio

### DISCUSSION THREAD

**Discussion by mihaigalos** [minio/minio#15337] [Q&A]
Hi,

What would happen if I upload a file and immediately download it from Minio (dockerized - minio/minio:latest - fc5011f7acaef5b7804ddef05e9a8c104f46f69c9676bcaae0cc49f520c69883 , linux, x64)?

Is there an internal request queue?
Is there any guarantee as to the sequencing of the `GET` after the `PUT` has finished?

I sometimes see some strange behaviour where the retrieved file's size is 0 (yielding a sha256sum of `e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855`).
This test^ is green when I run locally but sometimes I can see it happening in the CI.

[Logs](https://github.com/mihaigalos/aim/runs/7394061782?check_suite_focus=true) - Screenshot
![image](https://user-images.githubusercontent.com/16443090/179564894-e777eae9-87bc-4677-9abf-29574f713271.png)

**Comment by harshavardhana** [comment]
@mihaigalos docker `fc5011f7acaef5b7804ddef05e9a8c104f46f69c9676bcaae0cc49f520c69883` there is no image by this name.

**Reply by mihaigalos** [reply]
**Update: Became irrelevant.**

<details>
  <summary>Original reply</summary>

Sorry about that, I went through the provided logs again.
I see:
```
2022-07-18T16:50:46.1593032Z d2a2fb2e1191: Pull complete
2022-07-18T16:50:46.1596807Z Digest: sha256:5bc924071f16cdf61ee521a4b08e97eb6edc54f03deab767c9cbfba9a391ce75
2022-07-18T16:50:46.1597185Z Status: Downloaded newer image for minio/minio:latest
```
I however don't see the above digest in the [latest release tags](https://hub.docker.com/r/minio/minio/tags).
My CI job ran today at the above time CET (4h ago) and your  `minio:latest` was last pushed 21h ago with [sha256:bfe908e2f0e95863993d690c7aed6d0d523b82866d496b29fb87458338a1a59b](https://hub.docker.com/layers/minio/minio/latest/images/sha256-bfe908e2f0e95863993d690c7aed6d0d523b82866d496b29fb87458338a1a59b?context=explore), does this help, did I actually use the `bfe90` version?

</details>

**Reply by mihaigalos** [reply]
Anyway, I've put some delays between the PUT and the GET and the problem is still persistent.
This doesn't seem to be a `minio` problem. Thank you, I'll continue investigating here.

**Reply by harshavardhana** [reply]
MinIO writes all write with O_DIRECT - so if you are using some network fs, or FS that doesn't implement consistency guarantees then it may not work.
